roles/install/tasks/main.yml

```
@@ -0,0 +1,114 @@
+      }}
+
+- name: Include distribution specific variables
+  ansible.builtin.include_vars: '{{ lookup("ansible.builtin.first_found", params) }}'
+  vars:
+    params:
+      files:
+        - >-
+          D-{{
+            hostvars['localhost' if sops_install_on_localhost else inventory_hostname].ansible_facts.distribution
+          }}-{{
+            hostvars['localhost' if sops_install_on_localhost else inventory_hostname].ansible_facts.distribution_version
+          }}.yml
+        - >-
+          D-{{
+            hostvars['localhost' if sops_install_on_localhost else inventory_hostname].ansible_facts.distribution
+          }}-{{
+            hostvars['localhost' if sops_install_on_localhost else inventory_hostname].ansible_facts.distribution_major_version
+          }}.yml
+        - >-
+          D-{{
+            hostvars['localhost' if sops_install_on_localhost else inventory_hostname].ansible_facts.distribution
+          }}.yml
+        - >-
+          OS-{{
+            hostvars['localhost' if sops_install_on_localhost else inventory_hostname].ansible_facts.os_family
+          }}-{{ hostvars['localhost' if sops_install_on_localhost else inventory_hostname].ansible_facts.distribution_major_version }}.yml
+        - >-
+          OS-{{
+            hostvars['localhost' if sops_install_on_localhost else inventory_hostname].ansible_facts.os_family
+          }}.yml
+        - default.yml
+      paths:
+        - '{{ role_path }}/vars'
+
+- name: Auto-detect source to install sops from
+  ansible.builtin.include_tasks: detect_source.yml
+  when: sops_source == 'auto'
+
+- name: Install Mozilla sops from GitHub
+  ansible.builtin.include_tasks: github.yml
+  when: sops_source == 'github'
+
+- name: Install Mozilla sops from system package repositories
+  ansible.builtin.include_tasks: system.yml
+  when: sops_source == 'system'
+
+- name: Install system packages
+  ansible.builtin.package:
+    name: '{{ _community_sops_install_system_packages_actual }}'
+    allow_downgrade: '{{ true if _community_sops_install_allow_downgrade and sops_version != "latest" else omit }}'
+  become: '{{ sops_become_on_install }}'
+  delegate_to: '{{ "localhost" if sops_install_on_localhost else omit }}'
+  run_once: '{{ sops_install_on_localhost or omit }}'
+  when: _community_sops_install_system_packages_actual | length > 0
+
+- name: Install unsigned system packages
+  ansible.builtin.package:
+    name: '{{ _community_sops_install_system_packages_unsigned_actual }}'
+    allow_downgrade: '{{ true if _community_sops_install_allow_downgrade and sops_version != "latest" else omit }}'
+    disable_gpg_check: true
+  become: '{{ sops_become_on_install }}'
+  delegate_to: '{{ "localhost" if sops_install_on_localhost else omit }}'
+  run_once: '{{ sops_install_on_localhost or omit }}'
+  when: _community_sops_install_system_packages_unsigned_actual | length > 0
+
+- name: Install packages from URL/path (Debian)
+  ansible.builtin.package:
```

```suggestion
  ansible.builtin.apt:
```
would be better if you're using apt specific options. It still gets redirected either way, but this seems more sensible to me.

---

Implemented in 8b7bb46786a6ba45c425fb5511d5ef72351764d3.